content/2015.09.encrypter-les-donnees-en.rst

```diff
@@ -0,0 +1,101 @@
+End-to-end encryption using Kinto.js
+###########################################
+
+:date: 2015-09-14
+:summary: How to encrypt data before sending it to the server.
+:slug: encrypter-les-donnees
+:lang: en
+:url: encrypter-les-donnees
+
+A guest post this time, by Michiel de Jong from the Firefox OS team! :)
+
+As a first step in the Firefox OS
+"[Data Sync](https://wiki.mozilla.org/Firefox_OS_Data_Sync)" project, we're working on
+integrating Firefox OS with the Firefox Sync service that has already existed
+for years in Firefox Desktop, and which is also supported by Firefox Android
+and Firefox iOS. Adding Firefox OS to this family is an exciting challenge,
+because one of the philosophical goals of Firefox OS is to push the web as a
+platform for mobile apps, and we try to use the web platform wherever we can.
+
+Data Sync is no exception there, and it turns out that with a combination of
+[Kinto.js](https://github.com/Kinto/kinto.js) and
+[WebCrypto](http://www.w3.org/TR/WebCryptoAPI/), we were already able to build a
+[working proof-of-concept](https://www.youtube.com/watch?v=BEYgx5chun0)
+at the Next-Generation-Architecture workweek in Paris last month.
+
+To connect with the Firefox Sync Server, we use an in-between translating proxy
+called [Syncto](https://github.com/mozilla-services/syncto), which translates
+between the [Kinto protocol](http://kinto.readthedocs.org/en/latest/api/index.html)
+in the front, and the
+[SyncStorage protocol](https://docs.services.mozilla.com/storage/apis-1.5.html)
+in the back. The Firefox Accounts credentials are still obtained
+[by Gecko-layer code](https://developer.mozilla.org/en-US/docs/Firefox-Accounts-on-FirefoxOS),
+but we hope to move some of the code into the web-based Gaia-layer soon too.
+
+Data that is retrieved from Firefox Sync is decrypted as soon as it comes in,
+and stored unencrypted inside Kinto.js, for fast access and easy updating. When
+changes are pushed out to the remote server, it is encrypted again, just
+before it leaves the JavaScript sandbox.
+
+The way we tell Kinto.js that we want to use encryption, and provide our own
+"encode" and "decode" procedures, is quite simple. When creating a Kinto
+collection, we pass a RemoteTransformer as a collection-level option:
```

Comment:
Also, linking to the Transformers section from the docs might be useful to the reader here.